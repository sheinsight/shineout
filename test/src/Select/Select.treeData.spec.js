import { mount } from 'enzyme/build'
import React from 'react'
import TreeCase from '../../../site/pages/components/Select/example-15-treeData'

describe('Select[treeData]', () => {
  let wrapper
  const div = document.createElement('div')
  document.body.appendChild(div)
  beforeAll(() => {
    wrapper = mount(<TreeCase />, { attachTo: div })
  })
  test('should select when click options', () => {
    const getSelect = () => wrapper.find('ShineoutSelect').at(0)
    getSelect()
      .find('.so-select-inner')
      .simulate('click')
    expect(
      getSelect()
        .find('Result')
        .props().focus
    ).toBe(true)
  })
  test('should select when click options', () => {
    const getSelect = () => wrapper.find('ShineoutSelect').at(0)
    getSelect()
      .find('.so-select-inner')
      .simulate('click')
    getSelect()
      .find('.so-select-tree-node')
      .at(0)
      .simulate('click')
    wrapper.update()
    expect(
      getSelect()
        .find('Select')
        .props().value
    ).toBe('0')
  })
})
