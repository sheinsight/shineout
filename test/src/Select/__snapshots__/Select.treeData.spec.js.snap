// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select[treeData] should select when click options 1`] = `
<div>
  <div
    className="so-input so-input-inline so-select"
    style={
      Object {
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <div
      className="so-select-inner so-select-drop-down"
      data-id="select_l6uif38t"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      tabIndex={0}
    >
      <div
        className="so-select-result so-select-result-ltr so-select-empty"
      >
        <span
          className="so-input-placeholder so-select-ellipsis"
        >
          <span />
           
        </span>
        <a
          className="so-select-indicator so-select-caret"
          tabIndex={-1}
        >
          <svg
            viewBox="0 0 1024 1024"
          >
            <path
              d="M194.048 341.333333h636.842667c18.005333 0 30.592 23.125333 16.213333 39.125334l-311.253333 376.874666c-10.794667 14.208-34.133333 14.208-44.970667 0L176.085333 380.458667C163.498667 364.458667 174.293333 341.333333 194.048 341.333333z"
            />
          </svg>
        </a>
      </div>
      <div
        className="so-list so-hidable so-hidable-fade so-hidable-scale-y so-hidable-animation-240 so-select-options so-select-tree __hidable_l6uif38u__"
        data-id="select_l6uif38t"
        size="default"
        style={
          Object {
            "maxHeight": 250,
            "overflowY": "auto",
          }
        }
        type="default"
      >
        <span
          className="so-select-option so-select-option-ltr"
        >
          Data not found
        </span>
      </div>
    </div>
  </div>
  <br />
  <div
    className="so-input so-input-inline so-select"
    style={
      Object {
        "marginBottom": 20,
        "width": 250,
      }
    }
  >
    <div
      className="so-select-inner so-select-drop-down so-select-multiple"
      data-id="select_l6uif38x"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      tabIndex={0}
    >
      <div
        className="so-select-result so-select-result-ltr so-select-empty"
      >
        <span
          className="so-input-placeholder so-select-ellipsis"
        >
          <span />
           
        </span>
      </div>
      <div
        className="so-list so-hidable so-hidable-fade so-hidable-scale-y so-hidable-animation-240 so-select-options so-select-tree __hidable_l6uif38y__"
        data-id="select_l6uif38x"
        size="default"
        style={
          Object {
            "maxHeight": 250,
            "overflowY": "auto",
          }
        }
        type="default"
      >
        <span
          className="so-select-option so-select-option-ltr"
        >
          Data not found
        </span>
      </div>
    </div>
  </div>
</div>
`;
